src/main/java/org/kitteh/irc/client/library/implementation/IRCClient.java

```diff
@@ -103,9 +105,20 @@ protected void processElement(@Nonnull String element) {
     private final ActorProvider actorProvider = new ActorProvider(this);
 
     private Map<Character, ModeStatus<UserMode>> userModes;
+    private STSMachine stsMachine;
+    private boolean usingSts = false;
 
     IRCClient(@Nonnull Config config) {
         this.config = config;
+        if (this.config.get(Config.STS_STORAGE_MANAGER) != null) {
+            this.configureSts();
+        } else if (!this.isSSL()) {
+            throw new KittehAuthorNagException(
```

Comment:
From seeing the use here, I feel it's worth considering a rename to just `KittehNagException` because the author might not be the one configuring...